test/functional/repository/basic-methods/repository-basic-methods.ts

```diff
@@ -351,11 +369,187 @@ describe("repository > basic methods", () => {
 
             saved.id!.should.be.equal(1);
             saved.title.should.be.equal("New title");
-            saved.dateAdded.should.be.instanceof(Date);
-            saved.dateAdded.getTime().should.be.equal(date.getTime());
+            saved.dateAdded!.should.be.instanceof(Date);
+            saved.dateAdded!.getTime().should.be.equal(date.getTime());
         })));
     });
 
+    describe("upsert", function () {
+        function upsertableConnections(able: boolean | UpsertType) {
+            return function () {
+                return connections.filter((c) =>
+                    typeof able === "string"
+                        ? c.driver.supportedUpsertType === able
+                        : (c.driver.supportedUpsertType != null) === able
+                );
+            };
+        }
+        it("should first create then update an entity", asyncMapper(upsertableConnections(true), async (connection) => {
```

Comment:
we need to simplify code. Instead of quite complex for understanding `asyncMapper` and `upsertableConnections` we can have a simple checks like:

```ts
if (!connection.driver.supportedUpsertType) return
```

Reply:
incoming